src/Compilers/CSharp/Portable/FlowAnalysis/VariableUsePass.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// VariableUsePass marks all referenced variables in the visited node as used (as a read).
+    /// This pass is used for things like attribute arguments and default parameter expressions,
+    /// which are otherwise not analyzed for use references.
+    /// </summary>
+    internal class VariableUsePass : BoundTreeWalkerWithStackGuard
```

Comment:
>VariableUsePass [](start = 19, length = 15)

The name and the doc comment are confusing, this pass deals only with fields. Consider renaming and adjusting the comment.